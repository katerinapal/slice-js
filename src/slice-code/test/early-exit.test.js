import {runAllCombosTests, comboOfBools} from './helpers/utils'

// this is here to make it easy to isolate tests to a specific case
/*
test.only(
  'testing ifStatement(false)',
  require('./helpers/utils').snapSlice(
    require.resolve('./fixtures/early-exit'),
    ({ifStatement}) => {
      ifStatement(false)
    },
  ),
)
/* */

runAllCombosTests({
  filename: require.resolve('./fixtures/early-exit'),
  methods: [
    {methodName: 'ifStatement', possibleArguments: comboOfBools(1)},
    {methodName: 'condExpr', possibleArguments: comboOfBools(1)},
    {methodName: 'logicalExpr', possibleArguments: comboOfBools(1)},
  ],
})
