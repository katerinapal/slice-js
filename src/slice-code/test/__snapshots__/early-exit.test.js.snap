// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`condExpr(false) && condExpr(true) 1`] = `
"export {condExpr}

function condExpr(passIf) {
  if (!passIf) {
    return 'did not reach IfStatement'
  }
  return 'reached CondExpr Consequent'
}
"
`;

exports[`condExpr(false) 1`] = `
"export {condExpr}

function condExpr() {
  return 'did not reach IfStatement'
}
"
`;

exports[`condExpr(true) 1`] = `
"export {condExpr}

function condExpr() {
  return 'reached CondExpr Consequent'
}
"
`;

exports[`ifStatement(false) && ifStatement(true) 1`] = `
"export {ifStatement}

function ifStatement(passIf) {
  if (!passIf) {
    return 'did not reach IfStatement'
  }

  return 'reached IfStatement'
}
"
`;

exports[`ifStatement(false) 1`] = `
"export {ifStatement}

function ifStatement() {
  return 'did not reach IfStatement'
}
"
`;

exports[`ifStatement(true) 1`] = `
"export {ifStatement}

function ifStatement() {
  return 'reached IfStatement'
}
"
`;

exports[`logicalExpr(false) && logicalExpr(true) 1`] = `
"export {logicalExpr}

function logicalExpr(passIf) {
  if (!passIf) {
    return 'did not reach IfStatement'
  }
  return passIf && 'reached LogicalExpression'
}
"
`;

exports[`logicalExpr(false) 1`] = `
"export {logicalExpr}

function logicalExpr(passIf) {
  return 'did not reach IfStatement'
}
"
`;

exports[`logicalExpr(true) 1`] = `
"export {logicalExpr}

function logicalExpr(passIf) {
  return passIf && 'reached LogicalExpression'
}
"
`;
